contracts/cheddar_5x5_tic_tac_toe/src/lib.rs

```diff
@@ -388,70 +423,25 @@ impl Contract {
         game.change_state(GameState::Finished);
         self.internal_update_game(game_id, &game);
 
-        let game_to_store = GameLimitedView{
-            game_result: GameResult::Win(winner),
-            player1,
-            player2,
-            reward_or_tie_refund: GameDeposit {
-                token_id: game.reward().token_id,
-                balance
-            },
-            tiles: game.board.to_tiles(),
-        };
-
-        self.internal_store_game(game_id, &game_to_store);
-        assert_eq!(
-            game.game_state,
-            GameState::Finished,
-            "Cannot stop. Game in progress"
-        );
-        self.games.remove(game_id);
-    }
-
-    // TODO: remove this
-    pub fn stop_game(&mut self, game_id: &GameId) {
-        let mut game: Game = self.internal_get_game(&game_id);
-        assert_eq!(game.game_state, GameState::Active, "Current game isn't active");
-
-        let account_id = env::predecessor_account_id();
-        assert_ne!(env::predecessor_account_id(), game.current_player_account_id(), "No access");
-
-        let (player1, player2) = self.internal_get_game_players(game_id);
+        let last_move;
 
-        game.current_duration = env::block_timestamp() - game.initiated_at;
-        require!(
-            game.current_duration >= self.max_game_duration || env::block_timestamp() - game.last_turn_timestamp > self.max_turn_duration, 
-            "Too early to stop the game"
-        );
-
-        let (winner, looser) = if account_id == player1 {
-            (player1, player2)
-        } else if account_id == player2 {
-            (player2, player1)
+        if game.board.last_move.is_some() {
+            let last_move_piece = game.board.tiles.get(&game.board.last_move.clone().unwrap());
+            last_move = Some((game.board.last_move.clone().unwrap(), last_move_piece.unwrap()))
         } else {
-            panic!("You are not in this game. GameId: {} ", game_id)
-        };
-
-        self.internal_update_stats(
-            Some(&game.reward().token_id), 
-            &looser, 
-            UpdateStatsAction::AddPenaltyGame, 
-            None, 
-            None);
-
-        let balance = self.internal_distribute_reward(game_id, Some(&winner));
-        game.change_state(GameState::Finished);
-        self.internal_update_game(game_id, &game);
+            last_move = None;
+        }
 
         let game_to_store = GameLimitedView{
```

Comment:
lines from here to the end of the function are repeated in few other functions. Let's move them to a helper function,

Reply:
DONE